fortuna/prob_model/posterior/normalizing_flow/advi/advi_posterior.py

```diff
@@ -368,3 +368,10 @@ def unravel_fn(_params, _path):
             indices = np.concatenate((np.array([0]), np.cumsum(indices)))
 
         return rav, unravel, indices, rav_log_stds
+
+    def _decode_encoded_which_params(self, encoded_which_params: Tuple[Array]) -> Tuple[List[str, ...], ...]:
+        encoded_which_params = tree_map(lambda v: "".join([chr(o) for o in v]), encoded_which_params)
+        if isinstance(encoded_which_params, dict):
```

Comment:
Why can there be multiple formats for `encoded_which_params`? Isn't it always the same?

Reply:
Unfortunately, there is a difference if the state was stored in memory or on disk. When saving on disk and restoring, the encoded which params is transformed into a dictionary. So something like `([...], [...])` becomes `{'0': {'0': ...}, '1': {...}}`. Therefore I'm checking this by looking if the `encoded_which_params` is a dict.